Prevent changelog entries from being inserted to the end of file

AbhayVAshokan opened the pull request:
Fixes #6 
- Updated logic to insert new changelog entries below the "Unreleased" heading.

@Amaljith-K _a can you please share your thoughts about the same? Once resolved, I'll proceed to fix the tests. 

https://vimeo.com/752862375/fe4b010b6e

AbhayVAshokan committed:
Insert new entries below the "Unreleased" heading

Amaljith-K commented on app/Actions/PlaceReleaseNotesBelowUnreleasedHeadingAction.php:
```diff
@@ -31,7 +31,8 @@ public function __construct(
     public function execute(Heading $unreleasedHeading, string $latestCommit, string $headingText, string $releaseDate, ?string $releaseNotes, Document $changelog, string $compareUrlTargetRevision): Document
     {
-        $previousVersion = $this->getPreviousVersionFromUnreleasedHeading($unreleasedHeading);
+        // $previousVersion = $this->getPreviousVersionFromUnreleasedHeading($unreleasedHeading);
```
You can remove this line

Amaljith-K reviewed (commented):
@AbhayVAshokan _a I think it is best to throw an error if an unreleased heading is not found.
If possible make the message beginner friendly. Or otherwise, you can add an explanation on how to place an unreleased heading in the wiki & add a link in the error message.

AbhayVAshokan committed:
Throw exception when the 'Unreleased' heading does not exist

AbhayVAshokan committed:
Removed unreachable code

AbhayVAshokan committed:
Deleted irrelevant tests

AbhayVAshokan committed:
Deleted all content below unreleased heading in tests

AbhayVAshokan committed:
Updated build

AbhayVAshokan commented on app/Exceptions/UnreleasedHeadingDoesNotExistException.php:
```diff
@@ -0,0 +1,15 @@
+{
+    public function __construct()
+    {
+        parent::__construct("The 'Unreleased' heading was not found in the CHANGELOG. Please refer to the correct CHANGELOG format at: https://github.com/bigbinary/neeto-editor/blob/master/CHANGELOG.md.");
+    }
```
The URL shall be updated with the neetoKB article in neeto-engineering which contains a detailed explanation of how to setup the CHANGELOG file.

AbhayVAshokan commented:
@Amaljith-K _a please review. These are the new changes.
- Throw an exception when the "Unreleased" heading does not exist ([75e802c](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/75e802c4aed73c1cbef53f003ce446747c92ae1c)) and removed un-reachable code due to this change ([2e3ac75](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/2e3ac751d99bf74b3e24d543899eb9bcd0bdfead)).
- Deleted all irrelevant tests ([c9184e7](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/c9184e74ad5f04a862575b4789da673232238611)).
- Since the new logic directly inserts the entry under the "Unreleased" heading, there should be no content below the "Unreleased" heading and the next heading. Removed all such occurrences across tests ([5197f1f](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/5197f1fe9f98c61466dc03e6acc94688d5093328)).
- `composer build` command ([3c4cc7d](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/3c4cc7d3a2d79e0c13ab4e99b1a5e511c5dca79f)).

AbhayVAshokan committed:
Added link to neeto-engineering docs

Amaljith-K reviewed (approved):
LGTM

Amaljith-K commented:
@AbhayVAshokan raise a new PR in https://github.com/bigbinary/changelog-updater-action repo with the newly built bundle in it & assign it to me.
We can release that action as `v2` tag and use it like `bigbinary/changelog-updater-action@v2` since it has breaking changes.
